From 082ec87cd6bb4d42e43daf36adf0b0d319075fbf Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Thu, 29 Dec 2022 18:59:22 +0100
Subject: [PATCH 25/31] Revert "rtw88: Use rtw_iterate_vifs() for
 rtw_vif_watch_dog_iter()"

This reverts commit 56157ab2902231b106743e4c91403146fbe9a58d.

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 drivers/net/wireless/realtek/rtw88/main.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/net/wireless/realtek/rtw88/main.c b/drivers/net/wireless/realtek/rtw88/main.c
index fb4074e89..bcdf1f8c8 100644
--- a/drivers/net/wireless/realtek/rtw88/main.c
+++ b/drivers/net/wireless/realtek/rtw88/main.c
@@ -242,10 +242,8 @@ static void rtw_watch_dog_work(struct work_struct *work)
 	rtw_phy_dynamic_mechanism(rtwdev);
 
 	data.rtwdev = rtwdev;
-	/* rtw_iterate_vifs internally uses an atomic iterator which is needed
-	 * to avoid taking local->iflist_mtx mutex
-	 */
-	g
+	/* use atomic version to avoid taking local->iflist_mtx mutex */
+	rtw_iterate_vifs_atomic(rtwdev, rtw_vif_watch_dog_iter, &data);
 
 	/* fw supports only one station associated to enter lps, if there are
 	 * more than two stations associated to the AP, then we can not enter
-- 
2.39.0.rc2.1.gbd5df96b79

